go/test/endtoend/vtgate/queries/kill/kill_test.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kill
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/test/endtoend/utils"
+)
+
+// TestKillConnection kills its own connection and checks the error message received.
+func TestKillOwnConnection(t *testing.T) {
+	conn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	_, err = utils.ExecAllowError(t, conn, fmt.Sprintf("kill %d", conn.ConnectionID))
+	require.NoError(t, err)
+
+	// the connection should be closed.
+	_, err = utils.ExecAllowError(t, conn, "select 1")
+	require.ErrorContains(t, err, "EOF (errno 2013) (sqlstate HY000)")
+}
+
+// TestKillDifferentConnection kills different connection and check relevant error messages.
+func TestKillDifferentConnection(t *testing.T) {
+	conn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	killConn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer killConn.Close()
+
+	// connection does not exist
+	_, err = utils.ExecAllowError(t, killConn, "kill 12345")
+	require.ErrorContains(t, err, "Unknown thread id: 12345 (errno 1094) (sqlstate HY000)")
+
+	// connection exist
+	_, err = utils.ExecAllowError(t, killConn, fmt.Sprintf("kill %d", conn.ConnectionID))
+	require.NoError(t, err)
+
+	// executing on closed connection
+	_, err = utils.ExecAllowError(t, conn, "select 1")
+	require.ErrorContains(t, err, "EOF (errno 2013) (sqlstate HY000)")
+}
+
+// TestKillOwnQuery kills the kill statement itself
+func TestKillOwnQuery(t *testing.T) {
```

Comment:
do we want to keep this test? the assertion does not seem to assert anything

Reply:
It is asserting no error and also shows what it means to kill your own query.